src/Tasks/AssemblyDependency/ResolveAssemblyReference.cs

```diff
@@ -993,7 +993,7 @@ public ITaskItem[] SuggestedRedirects
         public ITaskItem[] FilesWritten
         {
             set { /*Do Nothing, Inputs not Allowed*/ }
-            get { return _filesWritten.ToArray(); }
+            get { return ReferenceTable.NormalizeAsArray(_filesWritten); }
```

Comment:
None of the normalization should run more than once per RAR execution except FilesWritten and UnresolvedAssemblyConflicts, which run once per access. My hunch is that that isn't an issue, but if it is, I'm happy to cache it so we only do that once.

Reply:
Let's cache it. Everything about RAR is a huge perf bottleneck.